client/my-sites/domains/domain-management/dns/index.jsx

```diff
@@ -20,20 +21,20 @@ import SectionHeader from 'components/section-header';
 import DnsTemplates from '../name-servers/dns-templates';
 import VerticalNav from 'components/vertical-nav';
 
-const Dns = React.createClass( {
-	propTypes: {
+class Dns extends React.Component {
+	static propTypes = {
 		domains: React.PropTypes.object.isRequired,
 		dns: React.PropTypes.object.isRequired,
 		selectedDomainName: React.PropTypes.string.isRequired,
 		selectedSite: React.PropTypes.oneOfType( [
 			React.PropTypes.object,
 			React.PropTypes.bool
 		] ).isRequired
-	},
+	};
 
-	getInitialState() {
-		return { addNew: true };
-	},
+	state = {
+		addNew: true
+	};
 
 	renderDnsTemplates() {
 		const selectedDomain = getSelectedDomain( this.props );
```

Comment:
Noticed this error in the console while testing this:
```
Uncaught TypeError: Cannot read property 'type' of undefined
    at isMappedDomain (index.js:118)
    at Dns.renderDnsTemplates (index.jsx:122)
    at Dns.render (index.jsx:172)
    at eval (ReactCompositeComponent.js:796)
    at measureLifeCyclePerf (ReactCompositeComponent.js:75)
    at ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext (ReactCompositeComponent.js:795)
    at ReactCompositeComponentWrapper._renderValidatedComponent (ReactCompositeComponent.js:822)
    at ReactCompositeComponentWrapper._updateRenderedComponent (ReactCompositeComponent.js:746)
    at ReactCompositeComponentWrapper._performComponentUpdate (ReactCompositeComponent.js:724)
    at ReactCompositeComponentWrapper.updateComponent (ReactCompositeComponent.js:645)
```

It's probably been introduced in https://github.com/Automattic/wp-calypso/pull/16518 (cc @delputnam), but we might as well fix it here. Seems that `selectedDomain` is null, because it probably did not load.

To reproduce: just refresh the DNS editor view 🤦‍♂️ Not sure how I did not notice that in my review 😞